Package.resolved

```diff
@@ -24,8 +24,8 @@
         "repositoryURL": "https://github.com/GigaBitcoin/secp256k1.swift.git",
         "state": {
           "branch": null,
-          "revision": "847f56717138cb3ae53896f1912b1f38a274d2fe",
```

Comment:
Any reason why this was changed? this is an important dependency to change a revision

Reply:
TBH I didn't change it manually, possibly Xcode changed it by itself. However it turns out that support for async/await is not as straightforward as I had thought, check this other PR https://github.com/vapor/vapor/pull/2883

So apparently Swift Concurrency was introduced in Swift 5.5, however at that point it was only available for macOS 12 and iOS 15, then on Swift 5.5.2 they've introduced support for async/await on macOS 10.15 and iOS 13. Therefore, the change proposed here will probably not build on Xcode 13.0 (Swift 5.5) or Xcode 13.1 (Swift 5.5.1), so is probably not worth merging anyway.

Having said that, I think the project as it is in `develop` might not be buildable by those versions of Xcode as well. I haven't tested it out though.

I'll be closing this PR since it probably needs more work around supporting different Xcode versions